static/app/components/workflowEngine/form/inlineNumberField.tsx

```diff
@@ -0,0 +1,18 @@
+import type {ComponentProps} from 'react';
+import styled from '@emotion/styled';
+
+import NumberField from 'sentry/components/forms/fields/numberField';
+
+type NumberFieldProps = ComponentProps<typeof NumberField>;
+
+export default function InlineNumberField(props: NumberFieldProps) {
+  return <StyledNumberField style={{height: '28px', minHeight: '28px'}} {...props} />;
```

Comment:
Adding the `flexibleControlStateSize` and `hideLabel` fields might help here. There might also be an `inline` prop? Basically, everything we need to set to get the fields normalized for the workflow engine use case should happen at this level.